### ISSUE TITLE
React TypeScript: `useForm`'s generic is forcing to use `type` where `interface` won't work

### ISSUE THREAD

**Issue by xsjcTony** [inertiajs/inertia#1624]
### Version:

- `@inertiajs/react` version: `v1.0.9`

### Describe the problem:

The following code won't work
```tsx
interface TestInterface {
  test: string
}

const _ = useForm<TestInterface>()
// Type 'TestInterface' does not satisfy the constraint 'Record<string, unknown>'.
// Index signature for type 'string' is missing in type 'TestInterface'.
```
while the following works
```tsx
type TestType = {
  test: string
};

const _ = useForm<TestType>()
// it works
```

### Steps to reproduce:

https://stackblitz.com/edit/vitejs-vite-9516uy?file=src%2FApp.tsx

### Suggestions

I don't see there's any reason to use `unknown` instead of `any` here, and I always prefer to use `interface` over `type` so it's frustrating for me. 

Change this ( `Record<string, unknown>` to `Record<string, any>` )
```ts
export default function useForm<TForm extends Record<string, unknown>>(initialValues?: TForm): InertiaFormProps<TForm>;
export default function useForm<TForm extends Record<string, unknown>>(rememberKey: string, initialValues?: TForm): InertiaFormProps<TForm>;
```
to 
```ts
export default function useForm<TForm extends Record<string, any>>(initialValues?: TForm): InertiaFormProps<TForm>;
export default function useForm<TForm extends Record<string, any>>(rememberKey: string, initialValues?: TForm): InertiaFormProps<TForm>;
```
solves the issue for me, just like in `react-hook-form`
```ts
export declare function useForm<TFieldValues extends FieldValues = FieldValues, TContext = any>(props?: UseFormProps<TFieldValues, TContext>): UseFormReturn<TFieldValues, TContext>;

export type FieldValues = Record<string, any>;
```

**Comment by Bloemendaal** [comment]
Possibly fixed by https://github.com/inertiajs/inertia/pull/1626

**Comment by reinink** [comment]
@xsjcTony Hey thanks for reporting this issue! While not a totally perfect fix (we've had to loosen the types a bit), this has been resolved in #1649 👍
